src/Scripting/AsTLTypes.hpp

```
@@ -0,0 +1,403 @@
+		F64 mag() const {
+			return std::sqrt(x * x + y * y + z * z);
+		}
+
+		// Vector normalization
+		VEC3 norm() const {
+			F64 magnitude = mag();
+			return { x / magnitude, y / magnitude, z / magnitude };
+		}
+	};
+	// Scalar addition with Left-side I32 (I32 + VEC3)
+	inline AsTL::VEC3 operator+(AsTL::F64 scalar, const AsTL::VEC3& v) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)